content/08-cardano-sidechains/03-example-evm-sidechain/04-deploy-smart-contracts/02-setup-development.mdx

```diff
@@ -0,0 +1,294 @@
+
+Our sample JavaScript project contains a `hardhat.config.js` file. The `networks` config field is an optional object where network names map to their configuration and we can set `accounts` we want to sign transactions. For example:
+
+```js
+require("@nomicfoundation/hardhat-toolbox");
+
+/** @type import('hardhat/config').HardhatUserConfig */
+
+const Private_Key = "INSERT_PRIV_KEY"
+
+module.exports = {
+  networks: {
+    evm-sidechain: {
+        url: `<PENDING_URL>`,
+        accounts: [`0x${Private_Key}`]
+    },
+  },
+  solidity: "0.8.17",
+};
+```
+
+## Step 03 - Compile and deploy
+
+- To compile Solidity files in the `contracts/` directory of your Hardhat project execute `npx hardhat compile`:
+
+```
+$ npx hardhat compile
+```
+
+Deploying with Hardhat utilizes deployment scripts located in `scripts/`. 
+
+- To deploy with Hardhat, execute the deploy script with `evm-sidechain` as the network option.
+
+```
+$ npx hardhat run scripts/deploy.js --network evm-sidechain
+Greeter deployed to: 0x95E1C8F6C00d8cB518faBA33c8Ae814fc12E3090
+```
+
+🎉 Hurray! You deployed your smart contract to EVM sidechain testnet and can search your deployment on the [testnet explorer]( <PENDING_URL>).
+
+# Truffle
+
+This tutorial will show you how to deploy a Solidity smart contract to the EVM sidechain testnet using Truffle. [Truffle](https://trufflesuite.com/truffle/) is a popular development framework for EVM-based applications.
+
+## Prerequisites
+
+Install the following dependencies:
+
+- [Node.js](https://github.com/nvm-sh/nvm) version 8 or later.
+- [Truffle Suite](https://trufflesuite.com/truffle/)
+
+Check Node.js and NPM are installed:
+
+```
+$ node --version
+$ npm --version
+```
+
+Install Truffle and check whether it installed:
+
+```
+$ npm install -g truffle
+$ truffle version
+```
+
+## Step 1 - Configure Truffle
+
+You need to create a Truffle project before configuring Truffle to deploy on the EVM sidechain testnet. This demonstration will use the [React Truffle Box](https://trufflesuite.com/boxes/react/).
+
+- Create a new project directory for your Truffle project and unbox the React Truffle Box:
+
+```
+$ mkdir React-Truffle
+$ cd React-Truffle
+$ truffle unbox react
+```
+
+- Configure Truffle to use the EVM sidechain testnet by adding the following in `React-Truffle/truffle/truffle-config.js` > Line 45. Remove the comment on Line 25.
+- Replace `YOUR_SEED_PHRASE` with your mnemonic phrase.
+
+## Step - Truffle Migrate
+
+At this point, you've defined the EVM sidechain testnet and your testnet account to sign smart contract deployments. You are now ready to deploy!
+
+- Within the `truffle/` directory of our project, execute `truffle migrate` while passing `evm-sidechain` for the `--network` parameter:
+
+```
+$ truffle migrate --network evm-sidechain
+```
+
+Upon executing `truffle migrate`, Truffle will return the transaction hash and contract address which can be searched on the EVM sidechain [testnet explorer](<PENDING_URL>).
```

Comment:
Upon executing `truffle migrate`, Truffle will return the transaction hash and contract address which can be searched on the EVM sidechain [testnet explorer](http://explorer.sidechain.evmtestnet.iohkdev.io/).